### PR TITLE
Adding app last RC/signal to monitoring registry

### PR DESCRIPTION
**Pull request by CatGarab**
Updating the monitoring registry to be persistent so that we can track the RC or signal from the last time an app was run.

### CODE REVIEW

File: services/app-service/src/monitor.rs [hunk 1]
```diff
@@ -16,76 +16,152 @@
...
+    name: &str,
+    run_level: &str,
+) -> Result<bool, AppError> {
+    let entries = registry.lock().map_err(|err| AppError::MonitorError {
+        err: format!("Failed get entries mutex: {:?}", err),
+    })?;
+
+    Ok(entries
+        .iter()
+        .any(|entry| entry.name == name && entry.run_level == run_level && entry.running))
```
[the remark this answers is not in the document]
Comment (CatGarab) [reply services/app-service/src/monitor.rs]: Yes. With our current setup, you'd have the OnBoot instance running some long-term logic, then the OnCommand instance for executing some one-off logic

File: services/app-service/src/monitor.rs [hunk 2]
```diff
@@ -16,76 +16,152 @@
...
-        .lock()
-        .map_err(|err| AppError::MonitorError {
-            err: format!(
-                "Failed to add {} to monitoring. Couldn't get entries mutex: {:?}",
-                entry.name, err
-            ),
-        })?
-        .push(entry.clone());
+    // Find the monitoring entry and update it with the return code/exit signal and the end time
+    if let Some(index) = entries
```
[the remark this answers is not in the document]
Comment (CatGarab) [reply services/app-service/src/monitor.rs]: I don't have strong feelings. Using a hashmap would be more efficient for lookup, but would be slightly slower for insert and require more storage. We're not going to be dealing with large numbers of apps, so I don't think there's much benefit either way.

File: services/app-service/src/monitor.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jacoffey3) [reply services/app-service/src/monitor.rs]: It will change when the runlevel is removed

File: services/app-service/tests/test_uninstall.rs [hunk 3]
```diff
@@ -320,3 +320,158 @@ fn uninstall_new_app() {
...
+            }"#,
+        );
+
+        assert!(result["appStatus"].as_array().unwrap().is_empty());
+    });
+
+    // Our app directory should now no longer exist
+    assert_eq!(fixture.registry_dir.path().join("dummy").exists(), false);
+
+    fixture.teardown();
```
Comment (plauche) [inline services/app-service/tests/test_uninstall.rs]: If you want you can implement `Drop` for the fixture and have it automatically run `teardown` when this test function returns. Also neat fact, rust will still run destructors of all items in scope even if a panic is raised.

Comment (CatGarab) [reply services/app-service/tests/test_uninstall.rs]: Ooooh, yeah, we should do that. But we should do that everywhere. [I made a card](https://trello.com/c/jA8p4h3g/1072-unit-tests-change-fixtureteardown-to-a-rust-drop-definition)

**Pull request merged**